plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -1070,21 +1071,26 @@ private static Optional<DataCompression> getTableDataCompression(Handle handle,
 
     private static Optional<DataCompression> getTableDataCompressionWithRetries(Handle handle, JdbcTableHandle table)
     {
-        // DDL operations can take out locks against system tables causing the `getTableDataCompression` query to deadlock
-        final int maxAttemptCount = 3;
-        RetryPolicy<Optional<DataCompression>> retryPolicy = new RetryPolicy<Optional<DataCompression>>()
+        return retryOnDeadlock(() -> getTableDataCompression(handle, table), "error when getting table compression info for '%s'".formatted(table));
+    }
+
+    public static <T> T retryOnDeadlock(CheckedSupplier<T> supplier, String exceptionMessageSuffix)
+    {
+        // DDL operations can take out locks against system tables causing queries against them to deadlock
+        int maxAttemptCount = 3;
```

Comment:
constant?

Reply:
Why, it's not reused anywhere and cannot be customized.

Reply:
just nit. up to you. 
not to initialize var. May be I would even want to inline it not to attract an attention, but it used twice.